src/index.js

```diff
@@ -1,567 +1,363 @@
+  return util.toChecksumAddress(addr)
 }
 
 
 
 /**
- * Combine hex-encoded shards together to reconstruct a secret.
- * @param  {Array of Array of strings}  shards a collection of hex-encoded
- *  shards
- * @return {string} the reconstructed secret
+ * Break a 384-bit ticket into three shards, any two of which can be used to
+ * recover it.
+ *
+ * Each shard is simply 2/3 of the ticket -- the first third, second third, and
+ * first and last thirds concatenated together.
+ *
+ * @param  {String} ticket a 384-bit @q ticket
+ * @return {Array<String>}
  */
-const combineHex = shards => {
-  const splat = shards.map(shard =>
-    splitAt(shard.length / 2, shard));
-  const buffers = splat.map(pair =>
-    pair.map(buf => Array.from(hex2buf(buf))));
-  const combined = combineBuffer(buffers);
-  return buf2hex(combined);
-}
+const shard = ticket => {
+  const ticketHex = ob.patq2hex(ticket)
+  const ticketBuf = Buffer.from(ticketHex, 'hex')
 
+  if (ticketBuf.length !== 48) {
+    return [ ticket ]
+  }
 
+  const shard0 = ticketBuf.slice(0, 32)
+  const shard1 = ticketBuf.slice(16)
+  const shard2 = Buffer.concat([
+    ticketBuf.slice(0, 16),
+    ticketBuf.slice(32)
+  ])
 
-/**
- * Combine @q-encoded shards together to reconstruct a secret.
- * @param  {Array of Array of strings}  shards a collection of @q-encoded
- *  shards
- * @return {string} the reconstructed secret
- */
-const combinePatq = shards => {
-  const hexed = shards.map(ob.patq2hex)
-  const max = lodash.reduce(hexed, (acc, hex) =>
-    hex.length > acc
-      ? hex.length
-      : acc
-    , 0)
-
-  const padded = hexed.map(hex => hex.padStart(max, '0'))
-
-  const combined = combineHex(padded)
-  return ob.hex2patq(combined)
+  return lodash.map([shard0, shard1, shard2], buf =>
+    ob.hex2patq(buf.toString('hex')))
 }
 
 
 
 /**
- * Convert a full wallet into a sharded wallet.  Transforms the owner's seed
- * into a number of shards, of which only a subset are required in order to
- * reconstruct the original.
+ * Generate an Urbit HD wallet given the provided configuration.
  *
- * @param  {object}  wallet full HD wallet
- * @return  {object} an object representing a sharded full HD wallet
+ * @param  {String}  ticket a 64, 128, or 384-bit @q master ticket
+ * @param  {Number}  ship an optional ship number
+ * @param  {String}  password a password used to salt generated seeds (default:
+ *   null)
+ * @param  {Number}  revision a revision number used as a salt (default: 0)
+ * @param  {Bool}  boot if true, generate network keys for the provided ship
+ *   (default: false)
+ * @return  {Promise<Object>}
  */
-const shardWallet = wallet => {
-  const walletCopy = lodash.cloneDeep(wallet);
-  const sharded = shardPatq(walletCopy.ticket)
-  walletCopy.ticket = sharded;
-  return walletCopy;
-}
+const generateWallet = async config => {
+  const { ticket } = config
+  const ship = 'ship' in config ? config.ship : null
```

Comment:
As with the salt, I think we default to `0` rather than complete absence. @msutherl ?

Reply:
Yes, this was the conclusion from the other day.